generator/transit_generator.hpp

```diff
@@ -102,11 +103,66 @@ class DeserializerFromJson
   OsmIdToFeatureIdsMap const & m_osmIdToFeatureIds;
 };
 
+/// \brief the data contains all the information to make TRANSIT_FILE_TAG section.
+class GraphData
+{
+public:
+  // @todo(bykoianko) All the methods below should be rewrite with the help of visitors.
+  void DeserializeFromJson(my::Json const & root, OsmIdToFeatureIdsMap const & mapping);
+  void SerializeToMwm(std::string const & mwmPath) const;
+  void Append(GraphData const & rhs);
+  void Clear();
+  bool IsValid() const;
+
+  void SortStops();
+  /// \brief Calculates and updates |m_edges| by adding valid value for Edge::m_weight
+  /// if it's not valid.
+  /// \note |m_stops|, Edge::m_stop1Id and Edge::m_stop2Id in |m_edges| should be valid before call.
+  void CalculateEdgeWeight();
+  /// \brief Calculates best pedestrian segment for every gate in |m_gates|.
+  /// \note All gates in |m_gates| should have a valid |m_point| field before the call.
+  void CalculateBestPedestrianSegment(string const & mwmPath, string const & countryId);
+  /// \brief Removes some items from all the class fields if they outside |mwmBorders|.
+  /// @todo(bykoinko) Certain rules which is used to clip the transit graph should be described here.
+  void ClipGraphByMwm(std::vector<m2::RegionD> const & mwmBorders);
```

Comment:
ClipGraph?

Reply:
s/mwmBorders/borders/
Am I right that your code somehow depends on the fact that mwmBorders are real borders for some mwm? Or this code works with any borders? If so, please, remove "mwm" prefix.

Reply:
This method was written not to depend from the fact if the border is real mwm borders on not. 

But after @tatiana-kondakova comments I decided to try to move the clipping step to generate features step of generator. In that case all transit graph mwm would be processed only once for build of planet. 

The idea is to take all transit jsons on the generate features step, to split them into json according to mwm borders, and to create transit and transit cross mwm sections on this stage.

What do you think about that?